cmd/osm-crd-conversion-webhook/osm-crd-conversion-webhook.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/openservicemesh/osm/pkg/crdconversion"
+	configClientset "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	"github.com/openservicemesh/osm/pkg/httpserver"
+	"github.com/openservicemesh/osm/pkg/k8s/events"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/metricsstore"
+	"github.com/openservicemesh/osm/pkg/signals"
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+var (
+	verbosity          string
+	kubeConfigFile     string
+	osmNamespace       string
+	caBundleSecretName string
+	osmMeshConfigName  string
+
+	crdConverterConfig crdconversion.Config
+
+	certProviderKind string
+
+	tresorOptions      providers.TresorOptions
+	vaultOptions       providers.VaultOptions
+	certManagerOptions providers.CertManagerOptions
+
+	scheme = runtime.NewScheme()
+)
+
+var (
+	flags = pflag.NewFlagSet(`osm-crd-conversion-webhook`, pflag.ExitOnError)
+	log   = logger.New("osm-crd-conversion-webhook/main")
+)
+
+func init() {
+	flags.StringVarP(&verbosity, "verbosity", "v", "info", "Set log verbosity level")
+	flags.StringVar(&kubeConfigFile, "kubeconfig", "", "Path to Kubernetes config file.")
+	flags.StringVar(&osmNamespace, "osm-namespace", "", "Namespace to which OSM belongs to.")
+	flags.StringVar(&osmMeshConfigName, "osm-config-name", "osm-mesh-config", "Name of the OSM MeshConfig")
+
+	// crd converter options
+	flags.IntVar(&crdConverterConfig.ListenPort, "webhook-port", 443, "Webhook port for crd conversion")
+
+	// Generic certificate manager/provider options
+	flags.StringVar(&certProviderKind, "certificate-manager", providers.TresorKind.String(), fmt.Sprintf("Certificate manager, one of [%v]", providers.ValidCertificateProviders))
+	flags.StringVar(&caBundleSecretName, "ca-bundle-secret-name", "", "Name of the Kubernetes Secret for the OSM CA bundle")
+
+	// Vault certificate manager/provider options
+	flags.StringVar(&vaultOptions.VaultProtocol, "vault-protocol", "http", "Host name of the Hashi Vault")
+	flags.StringVar(&vaultOptions.VaultHost, "vault-host", "vault.default.svc.cluster.local", "Host name of the Hashi Vault")
+	flags.StringVar(&vaultOptions.VaultToken, "vault-token", "", "Secret token for the the Hashi Vault")
+	flags.StringVar(&vaultOptions.VaultRole, "vault-role", "openservicemesh", "Name of the Vault role dedicated to Open Service Mesh")
+	flags.IntVar(&vaultOptions.VaultPort, "vault-port", 8200, "Port of the Hashi Vault")
+
+	// Cert-manager certificate manager/provider options
+	flags.StringVar(&certManagerOptions.IssuerName, "cert-manager-issuer-name", "osm-ca", "cert-manager issuer name")
+	flags.StringVar(&certManagerOptions.IssuerKind, "cert-manager-issuer-kind", "Issuer", "cert-manager issuer kind")
+	flags.StringVar(&certManagerOptions.IssuerGroup, "cert-manager-issuer-group", "cert-manager.io", "cert-manager issuer group")
+
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = admissionv1.AddToScheme(scheme)
```

Comment:
Passing cert-provider flags to all apps is just weird, although not specific to this change. As a separate effort, we should rethink the cert synchronization design.

Reply:
agree